extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -370,3 +350,34 @@ async function exists(path: string): Promise<boolean> {
 		return false;
 	}
 }
+
+/**
+ * processWhenClause takes in a when clause (either the word 'true' or a series of clauses in the format:
+ * '<type_name>=<value_name>' joined by '&&').
+ * If the where clause is true or undefined, return true as there is no clause to check.
+ * It evaluates each individual when clause by comparing the equivalent selected options (sorted in alphabetical order and formatted to match).
+ * If there is any selected option that doesn't match, return false.
+ * Return true if all clauses match.
+ */
+function processWhenClause(when: string, selectedOptions: { option: string, value: string }[]): boolean {
+	if (when === undefined || when.toString().toLowerCase() === 'true') {
+		return true;
+	} else {
+		const expected = when.replace(' ', '').split('&&').sort();
+		let actual: string[] = [];
+		selectedOptions.forEach(option => {
+			actual.push(`${option.option}=${option.value}`);
+		});
+		actual = actual.sort();
+
+		if (actual.length === expected.length) {
```

Comment:
So if I'm understanding this correctly - you require that a when clause contains ALL of the options that are passed here? That doesn't seem to be necessary to me - and just makes it more annoying to use. If I only need to switch on one of the option values then I shouldn't need to include any of the other possible options as well.